### PR TITLE
Use new API calls for OpenVino

### LINKED ISSUE #1083: openvino-rs 0.7.1 is yanked, new version has breaking changes
I found this weird issue after `cargo clean`, turns out 0.7.1 which we use is yanked (taken out altogether?).

The next version has a different API call to create Tensors, therefore `object_detection` is broken as of now (at least for non-nao builds).

- Release notes: https://github.com/intel/openvino-rs/releases/tag/v0.7.2
- The API change: https://github.com/intel/openvino-rs/commit/b6eacef9b5ab2cc964abbf02be3c74b0d7a948b1

```
error[E0599]: no function or associated item named `new_from_host_ptr` found for struct `Tensor` in the current scope
   --> crates/object_detection/src/pose_detection.rs:155:30
    |
155 |         let tensor = Tensor::new_from_host_ptr(
    |                              ^^^^^^^^^^^^^^^^^ function or associated item not found in `Tensor`
    |
note: if you're trying to build a new `Tensor` consider using one of the following associated functions:
      Tensor::new
      Tensor::from_ptr
   --> /home/darshana/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openvino-0.7.2/src/tensor.rs:21:5
    |
21  |     pub fn new(element_type: ElementType, shape: &Shape) -> Result<Self> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
33  |     pub(crate) fn from_ptr(ptr: *mut ov_tensor_t) -> Self {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

### PR DESCRIPTION
**Pull request by tuxbotix**
## Why? What?

Use the changed API for creating a tensor. How-To taken from their commit: https://github.com/intel/openvino-rs/commit/b6eacef9b5ab2cc964abbf02be3c74b0d7a948b1

Fixes #1083

## Ideas for Next Iterations (Not This PR)
- Add a 'smoke test' for the Object Detector to catch OpenVino setup, etc issues (Hopefully, the CI will also catch these?)

## How to Test

`./pepsi run` should build for webots and launch properly.

### CODE REVIEW

**Comment by oleflb** [comment]
unfortunately this requires cloning..., opened some issues over at openvino
- [133](https://github.com/intel/openvino-rs/issues/133)
- [134](https://github.com/intel/openvino-rs/issues/134)

**Comment by tuxbotix** [comment]
I think the reason to yank is the `0.7.1`, `0.7.0` releases have memory leaks and segfaults caused by the API that got replaced in addition to some others. 2-3 of these fixes are related to leaks and other memory issues -> https://github.com/intel/openvino-rs/releases/tag/v0.7.2 . While I'm not a fan of yanking a release, I can't blame them for doing this as the listed issues are quite serious.

The cost of copying is also discussed in the change of API (second link in my issue). IMHO a slightly less performant library is better than a library that might segfault.

Additionally, we can consider changing the cycler's architecture to keep the tensor as part of the state and load the input image directly into it instead of the scratchpad.

**Comment by oleflb** [comment]
No, not possible since the Tensor does not impl Send :(

File: crates/object_detection/src/pose_detection.rs [hunk 1]
```diff
@@ -152,11 +152,12 @@ impl PoseDetection {
         }
 
         let mut infer_request = self.network.create_infer_request()?;
-        let tensor = Tensor::new_from_host_ptr(
-            ElementType::F32,
-            &self.network.get_input()?.get_shape()?,
-            self.scratchpad.as_bytes(),
-        )?;
+        let mut tensor = Tensor::new(ElementType::F32, &self.network.get_input()?.get_shape()?)?;
```
Comment (oleflb) [inline crates/object_detection/src/pose_detection.rs:155]: remove `scratchpad` from node state and pass the tensor buffer to the `load_into_scratchpad` directly. This way we save ourselves a whole copy of the image

**Review by oleflb — CHANGES_REQUESTED** [review]

**Comment by tuxbotix** [comment]
Ouch! 😬 That's bad 😞.
I'll try to setup a benchmark today if I can to see the impact of the copy and some alternative we could try.

Based on the notes by the authors, they would need to touch a lot of code to get the lifetime information right to create a tensor from a pointer and ensure safety.

**Comment by oleflb** [comment]
For now we can just remove the `scratchpad` and create the `Tensor` in the cycle function. you can use `get_data_mut` to create a mutable f32 slice that you can pass to the load_into_scratchpad function.

File: crates/object_detection/src/pose_detection.rs (hunk 1 above)
Comment (tuxbotix) [reply crates/object_detection/src/pose_detection.rs]: Done, let me know if I should also remove the `AsBytes` trait as well.
I didn't previously notice why you used the scratchpad, it all makes sense :) .

In summary, we saved the copy but still have the memory allocation in each cycle.

**Comment by oleflb** [comment]
Yes, feel free to remove the trait. I'd say we merge this state with the allocation.

File: crates/object_detection/src/pose_detection.rs [hunk 2]
```diff
@@ -268,6 +255,8 @@ fn non_maximum_suppression(
     poses
 }
 
+// Allowing dead code for future use of `scratchpad` once it is restored.
+#[allow(dead_code)]
```
Comment (oleflb) [inline crates/object_detection/src/pose_detection.rs:259]: please remove instead. We can add this back (if needed)

**Review by oleflb — CHANGES_REQUESTED** [review]

File: crates/object_detection/src/pose_detection.rs (hunk 2 above)
Comment (tuxbotix) [reply crates/object_detection/src/pose_detection.rs]: Done :)

**Review by oleflb — APPROVED** [review]
thank you for fixing :)

**Comment by tuxbotix** [comment]
And thank you for integrating all of this and catching the things I missed!!

**Pull request merged**